tools/config/README.md

```diff
@@ -0,0 +1,44 @@
+# Common scalar build settings.
+
+## Overview
+
+These definitions are only for use in Bazel built-in tools.  In all
+other cases, use flag type definitions from github.com/bazelbuild/bazel-skylib.
+
+<a name="basic-example"></a>
+## Basic Example
+
+### //tools/my_pkg/BUILD
+
+```python
+load("//tools/config:common_settings.bzl", "bool_flag")
+
+bool_flag(
+    name = "experimental_new_behavior",
+    build_setting_default = False,
+    visibility = ["//visibility:public"],
```

Comment:
I feel like if you have this example here people will cargo cult including the visibility when they create build settings. Would you mind adding an explanation of why it might be useful?